_posts/jared_baker/2018-12-15-Bluestore.md

```diff
@@ -0,0 +1,171 @@
+---
+layout: post
+title:  "Migrating 8 PB of data from Filestore to Bluestore"
+breadcrumb: true
+author: jared_baker
+date: 2018-12-15
+categories: jared_baker
+tags:
+    - ceph
+    - linux
+    - bluestore
+    - tutorial
+teaser:
+    info: How we migrated an 8 PB cluster from Filestore to Bluestore
+    image: jared_baker/bluestore/Ceph_Logo_Standard_RGB_120411_fa.png
+header:
+    version: small
+    title: Software Engineering Blog
+    image: jared_baker/le-logo-standard.png
+    icon: icon-blog
+---
+## Introduction
+The Cancer Genome Collaboratory is a Cloud computing environment designed for cancer research and an official repository of 950 TB (and growing) worth of whole genomes for the [ICGC project](https://dcc.icgc.org/repositories). We have used OpenStack and Ceph to provide an extremely scalable and reliable infrastructure. Maintaining the pieces namely through upgrades is our most important and time consuming task in operating a cloud like this. Whether you are chasing new features, bug fixes, security patches or supportability it seems like we are always planning for the next upgrade.
+
+## About our Ceph environment
+Our Ceph cluster was born in 2014 and was built upon Ceph Giant and eventually upgraded to Hammer, then to Ceph Jewel and more recently this year to Luminous. Up until Luminous our OSD's were using the only available osd backend at the time, Filestore, with xfs and co-located journals and 3 replicas for data redundancy. Our number of OSD's has been growing steadily throughout the years to maximize purchase value per terabyte and spreading our budget over a number of years so that hardware purchases only occurred when capacity was forecasted.
```

Comment:
Consider writing out full acronym for OSD (Object Storage Daemon) in first occurrence, depending on audience level you're targeting.  For example, I as a newbie, did not know what OSD was reading this post the first time.